### PR TITLE
Update GitHub actions to use Node 16 and Ubuntu to latest

### PR DESCRIPTION
**Pull request by SDawley**
Signed-off-by: sdawley <sdawley@redhat.com>

### What does this PR do?
Update GitHub actions to use Node 16 and Ubuntu to latest

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/21763

### How to test this PR?

Run the github actions and verify they still work

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Development resource are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-development-resources)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by tolusha** [comment]
@SDawley 
Pls have a look at https://github.com/eclipse-che/che-operator/actions/runs/3653537634/jobs/6173111961

**Comment by nickboldt** [comment]
> @SDawley Pls have a look at https://github.com/eclipse-che/che-operator/actions/runs/3653537634/jobs/6173111961

```
Error: Unable to resolve action `docker/build-push-action@v6`, unable to find version `v6`
```

File: .github/workflows/pr-check.yml [hunk 1]
```diff
@@ -14,50 +14,50 @@ name: PR check
...
+      uses: actions/checkout@v3
     - name: Set up QEMU
-      uses: docker/setup-qemu-action@v1
+      uses: docker/setup-qemu-action@v2
     - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v1
+      uses: docker/setup-buildx-action@v2
     - name: Build images
-      uses: docker/build-push-action@v2
+      uses: docker/build-push-action@v6
```
Comment (nickboldt) [inline .github/workflows/pr-check.yml:44]:
```suggestion
      uses: docker/build-push-action@v3
```

**Review by nickboldt — APPROVED** [review]

**Review by tolusha — APPROVED** [review]

**Review by ibuziuk — APPROVED** [review]

**Comment by SDawley** [comment]
Conflicts resolved.

**Pull request merged**